modules/core/src/system.cpp

```diff
@@ -712,8 +712,12 @@ int64 getTickCount(void)
 #elif defined __MACH__ && defined __APPLE__
     return (int64)mach_absolute_time();
 #else
-    struct timeval tv;
+#if defined __QNX__
+    int tz;
+#else
     struct timezone tz;
+#endif
+    struct timeval tv;
     gettimeofday( &tv, &tz );
```

Comment:
[Documentation](http://www.qnx.com/developers/docs/6.5.0/index.jsp?topic=%2Fcom.qnx.doc.neutrino_lib_ref%2Fg%2Fgettimeofday.html) says that "This pointer must be NULL or the behavior of gettimeofday() is unspecified" for the second `void*` parameter.
So it is better do not pass pointer on int value on stack.

For both Linux (validated) / QNX this code should work well:
```
    struct timeval tv;
    gettimeofday(&tv, NULL);
```

Reply:
Yes, definitely agree. I considered that but was trying to make the most minimal change possible without having access to every supported architecture. But I checked this in the Linux implementation documentation and saw that they said the use of the timezone is actually now deprecated and NULL should be passed in general. http://man7.org/linux/man-pages/man2/gettimeofday.2.html

Looking more closely at what `gettimeofday` does, what it's doing is setting the timezone rather than using the timezone. This is probably why a non-null pointer might have undefined behavior, because it might write data to an undefined location in memory. Good catch!